common/lib/xmodule/xmodule/video_module/video_module.py

```diff
@@ -425,6 +425,23 @@ def editor_saved(self, user, old_metadata, old_content):
         This should be fixed too.
         """
         metadata_was_changed_by_user = old_metadata != own_metadata(self)
+
+        # There is an edge case when old_metadata and own_metadata are same and we are importing transcript from youtube
+        # then there is a syncing issue where html5_subs are not syncing with youtube sub, We can make sync better by
+        # checking if transcript is present for the video and if any html5_ids transcript is not present then trigger
+        # the manage_video_subtitles_save to create the missing transcript with particular html5_id.
+        if not metadata_was_changed_by_user and self.sub and hasattr(self, 'html5_sources'):
+            html5_ids = get_html5_ids(getattr(self, 'html5_sources', []))
```

Comment:
you've already checked that `self` has `html5_sources`, so you don't need `getattr` here